### PR TITLE
Sign In flow on server

### PR DESCRIPTION
**Pull request by GarryFirst-cripto**
Implement Sign In flow on the server
Сorrection of errors and comments.

### CODE REVIEW

**Comment by GarryFirst-cripto** [comment]
Api протестил.

File: backend/src/api/middlewares/authenticationMiddleware.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import { getCustomRepository } from 'typeorm';
+import { Request, Response, NextFunction } from 'express';
+import { compare } from '../../common/utils/encryptHelper';
+import UserRepository from '../../data/repositories/userRepository';
+
+const authenticationMiddleware = async (req: Request, res: Response, next: NextFunction) => {
```
Comment (AlikRakhmonov) [inline backend/src/api/middlewares/authenticationMiddleware.ts:6]: You were trying to use Passport so I expected that you would create a strategy. It's not a strategy so you can use kind of code in the service

File: backend/src/api/middlewares/authenticationMiddleware.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  try {
+    const { email, password } = req.body.user;
+    const mailText = email.toLowerCase();
+    const users = await getCustomRepository(UserRepository).getAll();
```
Comment (AlikRakhmonov) [inline backend/src/api/middlewares/authenticationMiddleware.ts:10]: it affects performance badly. You should get exact user from the DB directly

File: backend/src/api/middlewares/authenticationMiddleware.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    for (let i = 0; i < users.length; i += 1) {
+      const logg = await compare(password, users[i].password);
+      if (logg && users[i].email.toLowerCase() === mailText) {
+        req.body.user.id = users[i].id;
+        break;
+      }
+    }
+    res.sendDate = true;
```
Comment (AlikRakhmonov) [inline backend/src/api/middlewares/authenticationMiddleware.ts:18]: why do you need that?

File: backend/src/api/middlewares/authenticationMiddleware.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    next();
+  } catch (error) {
+    req.body.user = {};
```
Comment (AlikRakhmonov) [inline backend/src/api/middlewares/authenticationMiddleware.ts:21]: You use catch but you don't throw an error. It's not a good approach

File: backend/src/common/models/user/ILoginUser.ts [hunk 5]
```diff
@@ -0,0 +1,3 @@
+export interface ILoginUser {
+  id: string;
```
Comment (AlikRakhmonov) [inline backend/src/common/models/user/ILoginUser.ts:2]: I expect that ILoginUser contains email and password

File: backend/src/services/authService.ts [hunk 6]
```diff
@@ -17,3 +17,17 @@ export const register = async ({ password, ...userData }: IRegisterUser) => {
     token: createToken({ id: newUser.id })
   };
 };
+
+export const login = async ({ id }: ILoginUser) => {
```
Comment (AlikRakhmonov) [inline backend/src/services/authService.ts:21]: the previous comment didn't take into account

**Pull request closed without merging**